ykpack/src/types.rs

```diff
@@ -133,14 +131,65 @@ pub struct PlaceProjection {
 /// Describes a projection operation upon a projection base.
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
 pub enum ProjectionElem<V> {
-    Unimplemented(PhantomData<V>), // FIXME
+    Deref,
+    Field(FieldIndex),
+    Index(V),
+    ConstantIndex {
+        offset: u32,
+        min_length: u32,
+        from_end: bool,
+    },
+    Subslice {
+        from: u32,
+        to: u32,
+    },
+    Downcast(VariantIndex),
 }
 
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
 pub enum Operand {
     /// In MIR this is either Move or Copy.
     Place(Place),
-    Unimplemented, // FIXME constants
+    Constant(Constant),
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub enum Constant {
+    Scalar(Scalar),
+    Unimplemented,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub enum Scalar {
+    Bits {
```

Comment:
We've already spoken about this. We are going to use an enum for the different numeric types.

Notice how signedness isn't mentioned. I assume in MIR that'd be in the `Ty`, which we don't model.

So How about we have two numeric constant variants, signed and unsigned?

Reply:
I'm not too worried about how we do it at the moment, TBH: we're bound to learn from experience, so do something that seems sensible and we can iterate as we learn more.